loading.bs

```diff
@@ -0,0 +1,949 @@
+    the certificate can't be parsed enough to find this public key, return
+    `invalid`.
+1. If |publicKey|'s [=public key/algorithm=] is not [=id-ecPublicKey=] on the
+    [=secp256r1=] named curve, return `invalid`.
+1. If |parsedSignature|'s [=exchange signature/signature=] is not a valid
+    signature of |message| by |publicKey| using the [=ecdsa_secp256r1_sha256=]
+    algorithm, return `invalid`.
+1. Return `valid`.
+
+<h3 algorithm id="cross-origin-trust">Cross-origin trust</h3>
+
+A [=exchange signature/is valid|valid=] [=exchange signature=] |signature| <dfn
+for="exchange signature" lt="establishes cross-origin trust|does not establish
+cross-origin trust">establishes cross-origin trust</dfn> in an [=exchange=]
+|exchange| if the following steps return "trusted":
+
+1. Let |requestUrl| be |exchange|'s [=exchange/request=]'s [=request/url=].
+1. If |signature|'s [=exchange signature/validityUrl=]'s [=url/origin=] is not
+    [=same origin=] with |requestUrl|'s [=url/origin=], return "untrusted".
+1. If |exchange|'s [=exchange/request=]'s [=request/header list=] includes a
+    [=stateful request header=] or |exchange|'s [=exchange/response=]'s
+    [=response/header list=] includes a [=stateful response header=], return
+    "untrusted".
+1. If |signature|'s [=exchange signature/expiration time=] is more than 604800
+    seconds (7 days) after |signature|'s [=exchange signature/date=], return
+    "untrusted".
+1. If |signature|'s [=exchange signature/certificate chain=] [=certificate
+    chain/does not have a trusted leaf=] for |requestUrl|'s [=url/origin=],
+    return "untrusted".
+1. Return "trusted".
+
+<h3 algorithm id="trusting-certificate">Establishing trust in a certificate</h3>
+
+The [=certificate chain=] |chain| <dfn for="certificate chain" lt="has a trusted
+leaf|does not have a trusted leaf">has a trusted leaf</dfn> for an [=origin=]
+|origin| if the following steps return `trusted`:
+
+1. Let |authority| be |origin|'s [=origin/host=].
+1. Let |leaf| be |chain|'s [=certificate chain/leaf=].
+1. Attempt to build a trustworthy path from |leaf| to a trusted root using
+    [[!RFC5280]] and any other conventions used in making TLS ([[!RFC8446]])
+    connections. If no such path can be built, return `untrusted`.
+1. If |leaf|'s [=augmented certificate/certificate=] is not trusted for
+    |authority|, return `untrusted`.
+1. If |leaf|'s [=certificate/extensions=] don't map a [=CanSignHttpExchanges=]
+    OID to the ASN.1/DER encoding of NULL (0x05 0x00), return `untrusted`.
+1. If |leaf| has no [=augmented certificate/OCSP response=] or this response has
+    a lifetime of more than 7 days or isn't valid for |leaf|'s [=augmented
+    certificate/certificate=], return `untrusted`.
+
+    Note: This does not check for revocation of intermediate certificates, and
+    clients SHOULD implement another mechanism for that.
+
+1. If |leaf| doesn't have enough SCTs from trusted logs return `untrusted`,
```

Comment:
@sleevi Is this a reasonable way to phrase the requirement?

Reply:
No. There's no such thing as "trusted logs", and this also isn't how clients evaluate whether a certificate complies with a given CT policy.

I think much of this algorithm can be replaced by waving into the blackbox and saying that the OCSP response, the SCTs, and the certificates are inputs, for the host, and determine the response. I suspect you may want an added requirement on the OCSP responses freshness, or that the validating client MUST support Certificate Transparency (or is it optional for UAs that don't support CT but still support SXG?)

Reply:
You realize this means that only Chrome will implement SXGs for the foreseeable future? Coupling to CT in this way is going to either explicitly preclude certain UAs, or encourage them to adopt patterns that they've been reticent to do for TLS, precisely because they know it will harm the CT ecosystem. We should sync up on that. I realize the intersection between CT relates to SXG, but I think we need to be careful about MUSTing certain policy bits, compared to highlighting the tradeoffs.

Reply:
This is now a SHOULD with a link to a new Security Considerations section.